### PR TITLE
Unbound analysis for show statements

### PR DESCRIPTION
**Pull request by mkleen**
## Checklist

 - [x] User relevant changes are recorded in ``CHANGES.txt``
 - [x] Touched code is covered by tests
 - [x] Documentation has been updated if necessary
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

File: sql/src/main/java/io/crate/analyze/ShowStatementAnalyzer.java [hunk 1]
```diff
@@ -74,10 +75,11 @@ Query rewriteShowTransaction() {
...
 
     AnalyzedStatement analyzeShowTransaction(Analysis analysis) {
-        Query query = rewriteShowTransaction();
-        Analysis newAnalysis = analyzer.boundAnalyze(query, analysis.transactionContext(), ParameterContext.EMPTY);
-        analysis.rootRelation(newAnalysis.rootRelation());
-        return newAnalysis.analyzedStatement();
+        AnalyzedStatement analyzedStatement = analyzer.unboundAnalyze(rewriteShowTransaction(),
+                                                                      analysis.sessionContext(),
+                                                                      ParamTypeHints.EMPTY);
+        analysis.rootRelation((AnalyzedRelation) analyzedStatement);
```
Comment (mkleen) [inline sql/src/main/java/io/crate/analyze/ShowStatementAnalyzer.java]: Could maybe this cast be avoided somehow ?

**Comment by mkleen** [comment]
Retest this please

Running: Test class org.elasticsearch.cluster.coordination.ReconfiguratorTests

After 20s process did not stop

File: sql/src/main/java/io/crate/analyze/ShowStatementAnalyzer.java [hunk 2]
```diff
@@ -74,10 +75,11 @@ Query rewriteShowTransaction() {
     }
 
     AnalyzedStatement analyzeShowTransaction(Analysis analysis) {
-        Query query = rewriteShowTransaction();
-        Analysis newAnalysis = analyzer.boundAnalyze(query, analysis.transactionContext(), ParameterContext.EMPTY);
-        analysis.rootRelation(newAnalysis.rootRelation());
-        return newAnalysis.analyzedStatement();
+        AnalyzedStatement analyzedStatement = analyzer.unboundAnalyze(rewriteShowTransaction(),
```
Comment (seut) [inline sql/src/main/java/io/crate/analyze/ShowStatementAnalyzer.java]: could we use the `unboundAnalyze()`  helper method here as well?

File: sql/src/main/java/io/crate/analyze/ShowCreateTableAnalyzer.java [hunk 3]
```diff
@@ -35,9 +35,9 @@
         this.schemas = schemas;
     }
 
-    public ShowCreateTableAnalyzedStatement analyze(Table table, SessionContext sessionContext) {
+    public AnalyzedShowCreateTable analyze(Table table, SessionContext sessionContext) {
```
Comment (seut) [inline sql/src/main/java/io/crate/analyze/ShowCreateTableAnalyzer.java]: minor: maybe it make sense to move this into the `ShowStatementAnalyzer`

File: sql/src/main/java/io/crate/analyze/ShowStatementAnalyzer.java [hunk 4]
```diff
@@ -74,10 +82,16 @@ Query rewriteShowTransaction() {
     }
 
     AnalyzedStatement analyzeShowTransaction(Analysis analysis) {
-        Query query = rewriteShowTransaction();
-        Analysis newAnalysis = analyzer.boundAnalyze(query, analysis.transactionContext(), ParameterContext.EMPTY);
-        analysis.rootRelation(newAnalysis.rootRelation());
-        return newAnalysis.analyzedStatement();
+        return unboundAnalyze(rewriteShowTransaction(), analysis, ParamTypeHints.EMPTY);
```
Comment (seut) [inline sql/src/main/java/io/crate/analyze/ShowStatementAnalyzer.java]: why not use `unboundAnalyze(query, analysis)` and thus use `analysis.paramTypeHints()` here  as well?

Comment (mkleen) [reply sql/src/main/java/io/crate/analyze/ShowStatementAnalyzer.java]: The original method passed explicitly `ParamTypeHints.EMPTY` but I just see while debugging through that the analysis has anyway empty parameters, so this makes totally sense.

**Pull request merged**